File: migration/pset6/old/mario/more/mario.py
# mario.py

from cs50 import get_int
from cs50 import get_string

# prompts the user for the pyramid height
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Calculates initial amounts for space and pound characters
space = height - 1
pound = height - space

# draws the pyramid
for i in range(height):
    j = space
    k = pound

    # draws left half of pyramid
    while j > 0:
        print(" ", end="")
        j -= 1
    while k > 0:
        print("#", end="")
        k -= 1

    # draws the middle space
    print("  ", end="")

    # resets k to draw right half of pyramid
    k = pound
    while k > 0:
        print("#", end="")
        k -= 1
    print()

    # updates amounts of spaces and # chars to be drawn
    space -= 1
    pound += 1

File: CS50/pset1/old/mario.c
/*
 * mario.c
 *
 * This program prints a half
 * pyramid using spaces and hashes.
 * The height of the pyramid is determined
 * by the input entered by the user
 */
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    /*
     * Prompts the user for a positive integer
     * no greater than 23
     */
    int height;

    do
    {
        printf("Enter a height for your pyramid: ");
        height = get_int();
    }
    while ((height < 0) || (height > 23));

    // Determines the base of the pyramid
    int base = height + 1;

    /*
     * Calculates slots for spaces
     * and hashes
     */
    int space_slot = base - 2;
    int hash_slot = base - space_slot;

    // Draws the pyramid
    for (int row = 0; row < height; row++)
    {
        for (int space = 0; space < space_slot; space++)
        {
            printf(" ");
        }

        for (int hash = 0; hash < hash_slot; hash++)
        {
            printf("#");
        }

        printf("\n");

        /*
         * Decrements the spaces
         * and increments the hashes
         */
        space_slot--; hash_slot++;
    }
}